eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/OffsetProcessor.java

```diff
@@ -0,0 +1,88 @@
+package io.eventuate.local.mysql.binlog;
+
+import io.eventuate.common.eventuate.local.BinlogFileOffset;
+import io.eventuate.local.db.log.common.OffsetStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class OffsetProcessor {
+  protected Logger logger = LoggerFactory.getLogger(getClass());
+
+  private ConcurrentLinkedQueue<FutureOffset> futureOffsets = new ConcurrentLinkedQueue<>();
+  private AtomicBoolean processingOffsets = new AtomicBoolean(false);
+  private OffsetStore offsetStore;
+
+  public OffsetProcessor(OffsetStore offsetStore) {
+    this.offsetStore = offsetStore;
+  }
+
+  public void saveOffset(CompletableFuture<?> future, BinlogFileOffset binlogFileOffset) {
```

Comment:
As noted earlier: this should have a single parameter of type CompletableFuture<BinlogFileOffset>
I don't think there is any need for `FutureOffset`

Reply:
ok